src/schema.c

```diff
@@ -146,32 +300,31 @@ ldap_schema_append_attributetype(struct ldap_schema_t *schema, LDAPAttributeType
             ld_error("Attempt to pass NULL attribute type parameter. \n");
         }
 
+        if (!attributetype_names)
+        {
+            ld_error("Attribute type names list is empty!\n");
+        }
+
+        return false;
+    }
+
+    bool result = g_hash_table_insert(schema->attribute_types_by_oid, attributetype->at_oid, attributetype);
+
+    if (!result)
+    {
         return false;
     }
 
-    if (schema->attribute_types_size >= schema->attribute_types_capacity - 1)
+    for (int i = 0; attributetype_names[i] != NULL; ++i)
     {
-        int required_capacity = schema->attribute_types_capacity * 2;
-        TALLOC_CTX* ctx = talloc_parent(schema);
-        LDAPAttributeType** attributes = talloc_realloc(ctx,
-                                                        schema->attribute_types,
-                                                        LDAPAttributeType*,
-                                                        required_capacity);
-        if (!attributes)
+        result = result && g_hash_table_insert(schema->attribute_types_by_name, attributetype_names[i], attributetype);
```

Comment:
We do not need `result &&` here because we want to check result of the last operation here.